src/Avalonia.Styling/Styling/SelectorMatch.cs

```diff
@@ -5,51 +5,95 @@
 
 namespace Avalonia.Styling
 {
+    /// <summary>
+    /// Describes how a <see cref="SelectorMatch"/> matches a control and its type.
+    /// </summary>
+    public enum SelectorMatchResult
+    {
+        /// <summary>
+        /// The selector never matches this type.
+        /// </summary>
+        NeverThisType,
+
+        /// <summary>
+        /// The selector never matches this instance, but can match this type.
+        /// </summary>
+        NeverThisInstance,
+
+        /// <summary>
+        /// The selector always matches this type.
+        /// </summary>
+        AlwaysThisType,
+
+        /// <summary>
+        /// The selector always matches this instance, but doesn't always match this type.
+        /// </summary>
+        AlwaysThisInstance,
+
+        /// <summary>
+        /// The selector matches this instance based on the <see cref="SelectorMatch.Activator"/>.
+        /// </summary>
+        Sometimes,
+    }
+
     /// <summary>
     /// Holds the result of a <see cref="Selector"/> match.
     /// </summary>
     /// <remarks>
-    /// There are two types of selectors - ones whose match can never change for a particular
-    /// control (such as <see cref="Selectors.OfType(Selector, Type)"/>) and ones whose result can
-    /// change over time (such as <see cref="Selectors.Class(Selector, string)"/>. For the first
-    /// category of selectors, the value of <see cref="ImmediateResult"/> will be set but for the
-    /// second, <see cref="ImmediateResult"/> will be null and <see cref="ObservableResult"/> will
-    /// hold an observable which tracks the match.
+    /// A selector match describes whether and how a <see cref="Selector"/> matches a control, and
+    /// in addition whether the selector can ever match a control of the same type.
     /// </remarks>
     public class SelectorMatch
     {
-        public static readonly SelectorMatch False = new SelectorMatch(false);
+        /// <summary>
+        /// A selector match with the result of <see cref="SelectorMatchResult.NeverThisType"/>/
```

Comment:
nit: Extra '/' at the end of the line (throughout the doc comments in this file).